### ISSUE TITLE
Cannot start jupyter lab with --app-dir argument

### ISSUE THREAD

**Issue by yhlam** [jupyterlab/jupyterlab#10314]
## Description

The `jupyter lab` command doesn't recognize the `--app-dir` argument. The `--LabApp.app_dir` works though. According to the the output of `jupyter lab --help`, `--app-dir` should be equivalent to `--LabApp.app_dir`.

## Reproduce

1. Run `jupyter lab --app-dir=C:\Users\Public\myjupyterlabapp` command
2. See the error below
```
usage: jupyter-lab.EXE [-h] [--debug] [--generate-config] [-y] [--allow-root] [--no-browser] [--autoreload] [--script] [--no-script] [--core-mode] [--dev-mode] [--watch] [--expose-app-in-browser] [--extensions-in-dev-mode] [--log-level ServerApp.log_level]
                       [--config ServerApp.config_file] [--ip ServerApp.ip] [--port ServerApp.port] [--port-retries ServerApp.port_retries] [--transport KernelManager.transport] [--keyfile ServerApp.keyfile] [--certfile ServerApp.certfile]
                       [--client-ca ServerApp.client_ca] [--notebook-dir ServerApp.root_dir] [--browser ServerApp.browser] [--pylab ServerApp.pylab] [--gateway-url GatewayClient.url]
                       [extra_args [extra_args ...]]
jupyter-lab.EXE: error: unrecognized arguments: --app-dir=C:\Users\Public\myjupyterlabapp
```
3. Run `jupyter lab --LabApp.app_dir=C:\Users\Public\myjupyterlabapp` command, it works fine without error.

## Expected behavior

Launch JupyterLab with the app directory provided through the `--app-dir` argument.

## Context

- Operating System and version: Window 10
- Browser and version: Chrome
- JupyterLab version: 3.0.16

<details><summary>Troubleshoot Output</summary>
<pre>
$PATH:
        C:\Users\Public\virtualenvs\jpl-rBjss_p2-py3.8\Scripts
        C:\Users\username\.pyenv\pyenv-win\versions\3.8.6
        C:\Users\username\.pyenv\pyenv-win\versions\3.8.6\Scripts
        C:\Program Files (x86)\Common Files\Oracle\Java\javapath
        C:\Program Files (x86)\Support Tools\
        C:\Program Files (x86)\Windows Resource Kits\Tools\
        C:\WINDOWS\system32
        C:\WINDOWS
        C:\WINDOWS\System32\Wbem
        C:\WINDOWS\System32\WindowsPowerShell\v1.0\
        C:\WINDOWS\System32\OpenSSH\
        C:\Program Files (x86)\PuTTY\
        C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn\
        C:\Program Files\dotnet\
        C:\Program Files (x86)\dotnet\
        C:\Users\username\.pyenv\pyenv-win\bin
        C:\Users\username\.pyenv\pyenv-win\shims
        C:\Users\username\.poetry\bin
        C:\Users\username\scoop\shims
        C:\Users\username\AppData\Local\Microsoft\WindowsApps
        C:\Users\username\AppData\Local\Programs\Git\cmd
        C:\Users\Public\Programs\bin
        C:\Users\username\.local\bin
        C:\Users\Public\Programs\nodejs
        C:\Users\Public\Programs\chocolatey\bin
        C:\Users\Public\Programs\Microsoft VS Code\bin
        C:\Users\username\AppData\Local\Pandoc\
        C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Auxiliary\Build

sys.path:
        C:\Users\Public\virtualenvs\jpl-rBjss_p2-py3.8\Scripts\jupyter-troubleshoot.EXE
        c:\users\public\virtualenvs\jpl-rbjss_p2-py3.8\scripts\python38.zip
        C:\Users\username\.pyenv\pyenv-win\versions\3.8.6\DLLs
        C:\Users\username\.pyenv\pyenv-win\versions\3.8.6\lib
        C:\Users\username\.pyenv\pyenv-win\versions\3.8.6
        c:\users\public\virtualenvs\jpl-rbjss_p2-py3.8
        c:\users\public\virtualenvs\jpl-rbjss_p2-py3.8\lib\site-packages
        c:\users\public\virtualenvs\jpl-rbjss_p2-py3.8\lib\site-packages\win32
        c:\users\public\virtualenvs\jpl-rbjss_p2-py3.8\lib\site-packages\win32\lib
        c:\users\public\virtualenvs\jpl-rbjss_p2-py3.8\lib\site-packages\Pythonwin

sys.executable:
        c:\users\public\virtualenvs\jpl-rbjss_p2-py3.8\scripts\python.exe

sys.version:
        3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)]

platform.platform():
        Windows-10-10.0.18362-SP0

where jupyter:
        C:\Users\Public\virtualenvs\jpl-rBjss_p2-py3.8\Scripts\jupyter.exe

pip list:
        Package             Version
        ------------------- -------
        anyio               3.1.0
        argon2-cffi         20.1.0
        async-generator     1.10
        atomicwrites        1.4.0
        attrs               21.2.0
        Babel               2.9.1
        backcall            0.2.0
        bleach              3.3.0
        cffi                1.14.5
        colorama            0.4.4
        decorator           5.0.9
        defusedxml          0.7.1
        entrypoints         0.3
        idna                3.2
        ipykernel           5.5.5
        ipython             7.24.0
        ipython-genutils    0.2.0
        jedi                0.18.0
        Jinja2              3.0.1
        json5               0.9.5
        jsonschema          3.2.0
        jupyter-client      6.2.0
        jupyter-core        4.7.1
        jupyter-server      1.8.0
        jupyterlab          3.0.16
        jupyterlab-pygments 0.1.2
        jupyterlab-server   2.5.2
        MarkupSafe          2.0.1
        matplotlib-inline   0.1.2
        mistune             0.8.4
        more-itertools      8.8.0
        nbclassic           0.3.1
        nbclient            0.5.3
        nbconvert           6.0.7
        nbformat            5.1.3
        nest-asyncio        1.5.1
        notebook            6.4.0
        packaging           20.9
        pandocfilters       1.4.3
        parso               0.8.2
        pickleshare         0.7.5
        pip                 21.1.1
        pluggy              0.13.1
        prometheus-client   0.10.1
        prompt-toolkit      3.0.18
        py                  1.10.0
        pycparser           2.20
        Pygments            2.9.0
        pyparsing           2.4.7
        pyrsistent          0.17.3
        pytest              5.4.3
        python-dateutil     2.8.1
        pytz                2021.1
        pywin32             301
        pywinpty            1.1.1
        pyzmq               22.1.0
        requests            2.15.1
        Send2Trash          1.5.0
        setuptools          56.0.0
        six                 1.16.0
        sniffio             1.2.0
        terminado           0.10.0
        testpath            0.5.0
        tornado             6.1
        traitlets           5.0.5
        wcwidth             0.2.5
        webencodings        0.5.1
        websocket-client    1.0.1
        wheel               0.36.2

</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
usage: jupyter-lab.EXE [-h] [--debug] [--generate-config] [-y] [--allow-root] [--no-browser] [--autoreload] [--script]
                       [--no-script] [--core-mode] [--dev-mode] [--watch] [--expose-app-in-browser]
                       [--extensions-in-dev-mode] [--log-level ServerApp.log_level] [--config ServerApp.config_file]
                       [--ip ServerApp.ip] [--port ServerApp.port] [--port-retries ServerApp.port_retries]
                       [--transport KernelManager.transport] [--keyfile ServerApp.keyfile]
                       [--certfile ServerApp.certfile] [--client-ca ServerApp.client_ca]
                       [--notebook-dir ServerApp.root_dir] [--browser ServerApp.browser] [--pylab ServerApp.pylab]
                       [--gateway-url GatewayClient.url]
                       [extra_args [extra_args ...]]
jupyter-lab.EXE: error: unrecognized arguments: --app-dir=C:\Users\Public\myjupyterlabapp
</pre>
</details>

**Comment by fcollonval** [comment]
I track the error to that line:

https://github.com/jupyter-server/jupyter_server/blob/68a64ea13be5d0d86460f04e0c47eb0b6662a0af/jupyter_server/extension/application.py#L486

This is calling the `ServerApp.initialize` that will call `JupyterApp.initialize` in which the cli will be parsed and raise the error.

**Comment by fcollonval** [comment]
This has been fixed https://github.com/jupyter-server/jupyter_server/pull/564 and available in [`jupyter_server` 1.10.2](https://github.com/jupyter-server/jupyter_server/releases/tag/v1.10.2).
